internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -250,15 +250,18 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 		//
 		secondMachineSet := machineSets.Items[0]
 		t.Log("Scaling the MachineDeployment to 3 replicas")
-		modifyFunc := func(d *clusterv1.MachineDeployment) { d.Spec.Replicas = pointer.Int32(3) }
```

Comment:
Before this change:
The test only worked because the `updateMachineDeployment` in l.257 implicitly updated the replica field in deployments to 3. As a result l.396 created enough fake nodes (3) so that the additional MS got deleted which was then verified by l.402++

By dropping the side effect in updateMachineDeployment, the replica field in deployment stayed at 2. Thus we didn't create enough fake nodes and the old MS couldn't be cleaned up.